web/src/components/instances/InstanceForm.tsx

```diff
@@ -99,52 +97,58 @@ export function InstanceForm({ instance, onSuccess, onCancel }: InstanceFormProp
       </form.Field>
 
       <form.Field
-        name="host"
+        name="url"
         validators={{
           onChange: ({ value }) => {
-            if (!value) return 'Host is required'
-            if (!value.match(/^https?:\/\//)) return 'Host must start with http:// or https://'
-            return undefined
-          },
-        }}
-      >
-        {(field) => (
-          <div className="space-y-2">
-            <Label htmlFor={field.name}>Host URL</Label>
-            <Input
-              id={field.name}
-              value={field.state.value}
-              onBlur={field.handleBlur}
-              onChange={(e) => field.handleChange(e.target.value)}
-              placeholder="http://localhost"
-            />
-            {field.state.meta.isTouched && field.state.meta.errors[0] && (
-              <p className="text-sm text-destructive">{field.state.meta.errors[0]}</p>
-            )}
-          </div>
-        )}
-      </form.Field>
-
-      <form.Field
-        name="port"
-        validators={{
-          onChange: ({ value }) => {
-            if (!value || value < 1 || value > 65535) {
-              return 'Port must be between 1 and 65535'
+            if (!value) return 'URL is required'
+            
+            try {
+              // Parse URL, adding http:// if no protocol specified
+              const url = new URL(value.includes('://') ? value : `http://${value}`)
+              
+              // Validate protocol
+              if (url.protocol !== 'http:' && url.protocol !== 'https:') {
+                return 'Only HTTP and HTTPS protocols are supported'
+              }
+              
+              // Check if hostname is an IP address
+              const hostname = url.hostname
+              
+              // Proper IPv4 validation - each octet must be 0-255
+              const isIPv4 = /^(\d{1,3}\.){3}\d{1,3}$/.test(hostname) && 
+                            hostname.split('.').every(octet => {
+                              const num = parseInt(octet, 10)
+                              return num >= 0 && num <= 255
+                            })
+              
+              // IPv6 addresses are wrapped in brackets by URL parser
+              const isIPv6 = hostname.startsWith('[') && hostname.endsWith(']')
+              
+              // localhost doesn't require a port (qBittorrent default is 8080)
+              const isLocalhost = hostname === 'localhost' || hostname === '127.0.0.1'
+              
+              // Require port for IP addresses (except localhost)
+              // This helps prevent common mistakes where users forget the port
+              if ((isIPv4 || isIPv6) && !isLocalhost && !url.port) {
+                return 'Port is required when using an IP address (e.g., :8080)'
+              }
+              
+              return undefined
+            } catch {
+              return 'Please enter a valid URL'
```

Comment:
You could utilize validation with Zod. It has a `ZodURL` type which should work here, I think.